ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -40,26 +40,35 @@ const PanelDrawer = () => {
   const { panels, updatePanel } = usePanels();
   const { panelDrawer, closePanelDrawer } = useDashboardApp();
 
-  let defaultPanelName = '';
-  let defaultDescription = '';
-  if (panelDrawer?.panelKey) {
-    // editing an existing panel
-    defaultPanelName = panels[panelDrawer.panelKey]?.spec.display.name ?? '';
-    defaultDescription = panels[panelDrawer.panelKey]?.spec.display.description ?? '';
-  }
-  const [group, setGroup] = useState(panelDrawer?.groupIndex);
+  const panelKey = panelDrawer?.panelKey;
+  const panelGroupIndex = panelDrawer?.groupIndex;
+  const panelDefinition = panelKey ? panels[panelKey] : undefined;
+
+  const defaultGroup = panelGroupIndex ?? 0;
+  const defaultPanelName = panelDefinition?.spec.display.name ?? '';
+  const defaultPanelDescription = panelDefinition?.spec.display.description ?? '';
+  const defaultKind = panelDefinition?.spec.plugin.kind ?? '';
+  const defaultOptions = panelDefinition?.spec.plugin.spec;
+
+  const [group, setGroup] = useState(defaultGroup);
   const [panelName, setPanelName] = useState(defaultPanelName);
-  const [panelDescription, setPanelDescription] = useState(defaultDescription);
-  const [kind, setKind] = useState('');
-  const [options, setOptions] = useState<unknown>({});
+  const [panelDescription, setPanelDescription] = useState(defaultPanelDescription);
+  const [kind, setKind] = useState(defaultKind);
+  // Plugin options depends on the plugin kind ('LineChart', 'Markdown', et cetera).
+  // As the user switches between kinds, we keep track of the options
+  // so that we can go back to wherever they left off.
+  const [kindToOptions, setKindToOptions] = useState<Record<string, unknown>>({ [defaultKind]: defaultOptions });
```

Comment:
Oooh, nice. That's an interesting idea and definitely a better experience than the current one. 👍

Reply:
Yay, glad you like. It was kind of a necessity too, because the createInitialOptions() would overwrite the initial data otherwise 😄 We had a similar thing at my last job and we did the "keep track of current kind and compare when it changes" solution, but I think I like this way more.

Reply:
+1 really like this idea!